src/bp/core/services/dialog/decision-engine.ts

```diff
@@ -35,7 +34,39 @@ export class DecisionEngine {
     }
   }
 
-  private async sendSuggestedReply(reply, sessionId, event) {
+  protected _findBestReply(event: IO.IncomingEvent): IO.SuggestedReply | undefined {
+    const replies = _.sortBy(event.suggestedReplies, reply => -reply.confidence)
+    const lastMsg = _.last(event.state.session.lastMessages)
+
+    // If the user asks the same question, chances are he didnt get the response he wanted.
+    // So we cycle through the other suggested replies and return the next best reply with a high enough confidence.
+    for (let i = 0; i < replies.length; i++) {
+      const bestReplyIntent = replies[i].intent
+      const lastMessageIntent = lastMsg && lastMsg.intent
+
+      if (bestReplyIntent === lastMessageIntent) {
+        const nextBestReply = replies[i + 1]
+
+        if (this._isConfidentReply(nextBestReply)) {
+          return nextBestReply
+        } else {
+          return // If confidence is too low, we dont need to check other replies
+        }
+      }
+    }
+
+    const bestReply = replies[0]
+    if (this._isConfidentReply(bestReply)) {
+      return bestReply
+    }
+    return
+  }
+
+  private _isConfidentReply(reply) {
+    return reply && reply.confidence > this.MIN_CONFIDENCE
```

Comment:
`>=` ?

Reply:
sure. Also should we increase the min confidence level to something like 0.55?

Reply:
Confidence should be in the configuration, or use the NLU module config.